pkg/server/requestinfo/kcp_request_info_resolver_test.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2023 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package requestinfo
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestRequestInfoResolver(t *testing.T) {
+	tests := map[string]struct {
+		path                string
+		expectedPath        string
+		expectedResource    string
+		expectedSubresource string
+	}{
+		"path with /api prefix": {
+			path:             "/api/v1/namespaces/default/pods",
```

Comment:
Why are these valid paths? You need to prefix by cluster or service, there's no default logical cluster to access without specifying.

Reply:
at some point we redirected to `system:admin`, but disabled or broke that behaviour. Now the behaviour is a nasty error messages in the log. We should decide how to move forward and make it an hard error early in the stack, e.g. here.

Reply:
Btw, these tests here might (or not, to be verifiy) actually be correct. In `WithInClusterServiceAccountRequestRewrite` we handle the case of a service account calling back without the `/cluster` prefix, but with the cluster set in the JWT.

Reply:
We also strip the `/clusters/..` prefix here before passing to the RequestInfoResolver:

https://github.com/kcp-dev/kcp/blob/88d6071b2aaf47bcd5e854f8209e4c3bd8035b35/pkg/cache/server/config.go#L172-L186

Reply:
if we strip the cluster prefix, why do we need this special requestinfo resolver in the first place?